### PR TITLE
(fleet/*) migrate prometheus crds to prometheus-operator-crds

### PR DESCRIPTION
**Pull request by jhoblitt**
Depends on #281

### CODE REVIEW

File: fleet/lib/kyverno-conf/servicemonitor-kyverno-metrics.yaml [hunk 1]
```diff
@@ -6,7 +6,6 @@ metadata:
   namespace: kyverno
   labels:
     o11y.eu/monitor: enabled
```
Comment (fbegyn) [inline fleet/lib/kyverno-conf/servicemonitor-kyverno-metrics.yaml:8]: I though the consensus was `lsst.io/monitor: "true"`? If so, please change.

File: fleet/lib/mimir/overlays/pillan/values.yaml [hunk 2]
```diff
@@ -9,18 +9,15 @@ metaMonitoring:
     enabled: true
     labels:
       o11y.eu/monitor: enabled
```
Comment (fbegyn) [inline fleet/lib/mimir/overlays/pillan/values.yaml:11]: `lsst.io/monitor: "true"`? If so, please change.

File: fleet/lib/mimir/overlays/pillan/values.yaml [hunk 3]
```diff
@@ -9,18 +9,15 @@ metaMonitoring:
...
-      release: kube-prometheus-stack
       grafana_dashboard: "1"
   serviceMonitor:
     enabled: true
     labels:
       o11y.eu/monitor: enabled
```
Comment (fbegyn) [inline fleet/lib/mimir/overlays/pillan/values.yaml:16]: `lsst.io/monitor: "true"`? If so, please change.

File: fleet/lib/mimir/overlays/pillan/values.yaml [hunk 4]
```diff
@@ -9,18 +9,15 @@ metaMonitoring:
...
-      release: kube-prometheus-stack
   prometheusRule:
     enabled: true
     labels:
       o11y.eu/monitor: enabled
```
Comment (fbegyn) [inline fleet/lib/mimir/overlays/pillan/values.yaml:20]: `lsst.io/monitor: "true"`? If so, please change.

File: fleet/lib/mimir/overlays/ruka/values.yaml [hunk 5]
```diff
@@ -9,18 +9,15 @@ metaMonitoring:
     enabled: true
     labels:
       o11y.eu/monitor: enabled
```
Comment (fbegyn) [inline fleet/lib/mimir/overlays/ruka/values.yaml:11]: `lsst.io/monitor: "true"`? If so, please change.

File: fleet/lib/mimir/overlays/ruka/values.yaml [hunk 6]
```diff
@@ -9,18 +9,15 @@ metaMonitoring:
...
-      release: kube-prometheus-stack
       grafana_dashboard: "1"
   serviceMonitor:
     enabled: true
     labels:
       o11y.eu/monitor: enabled
```
Comment (fbegyn) [inline fleet/lib/mimir/overlays/ruka/values.yaml:16]: `lsst.io/monitor: "true"`? If so, please change.

File: fleet/lib/mimir/overlays/ruka/values.yaml [hunk 7]
```diff
@@ -9,18 +9,15 @@ metaMonitoring:
...
-      release: kube-prometheus-stack
   prometheusRule:
     enabled: true
     labels:
       o11y.eu/monitor: enabled
```
Comment (fbegyn) [inline fleet/lib/mimir/overlays/ruka/values.yaml:20]: `lsst.io/monitor: "true"`? If so, please change.

**Review by fbegyn — COMMENTED** [review]
Might be that I missed it in the prometheus deployment, but switching to `lsst.io/monitor` and `lsst.io/discover` on the selectors was the concensus I thought?

File: fleet/lib/kyverno-conf/servicemonitor-kyverno-metrics.yaml (hunk 1 above)
Comment (jhoblitt) [reply fleet/lib/kyverno-conf/servicemonitor-kyverno-metrics.yaml:8]: Updating the existing scrape labels is not the scope of this PR.  The goal is to decouple the prometheus CRDs from the prometheus deployment to resolve the circular dependency problem of the prometheus sts depending on `cm`s which come from the `snmp-exporter` deployment.  The string `kube-prometheus-stack` was removed where it wasn't required to make auditing the change easier. A double benefit to this change is allowing most other deployments to proceed even if the prometheus deployment hasn't completed.

All of these labels will need to be changed at the same time:

```
$ ack o11y.eu/monitor
strimzi-kafka-dashboards/fleet.yaml
6:  o11y.eu/monitor: enabled

mimir-pre/fleet.yaml
6:  o11y.eu/monitor: enabled

external-secrets/fleet.yaml
6:  o11y.eu/monitor: enabled

rook-ceph/fleet.yaml
6:  o11y.eu/monitor: enabled

kyverno-conf/servicemonitor-kyverno-metrics.yaml
8:    o11y.eu/monitor: enabled

kube-prometheus-stack/fleet.yaml
6:  o11y.eu/monitor: enabled

kyverno/fleet.yaml
7:  o11y.eu/monitor: enabled

mimir/overlays/ruka/values.yaml
11:      o11y.eu/monitor: enabled
17:      o11y.eu/monitor: enabled
22:      o11y.eu/monitor: enabled

mimir/overlays/pillan/values.yaml
11:      o11y.eu/monitor: enabled
17:      o11y.eu/monitor: enabled
22:      o11y.eu/monitor: enabled

mimir/fleet.yaml
7:  o11y.eu/monitor: enabled
```

Comment (jhoblitt) [reply fleet/lib/kyverno-conf/servicemonitor-kyverno-metrics.yaml:8]: Also, this label will need to be updated on resource managed by argo and the person responsible is on vacation until next week.

**Review by fbegyn — APPROVED** [review]
LGTM

**Pull request merged**